### DISCUSSION TITLE
How to refresh the diagram after programmatically adding nodes

### DISCUSSION THREAD

**Discussion by mjcee** [xyflow/xyflow#4733] [Q&A] [answered]
I am using something similar to the React Flow Dagre Tree example program - https://reactflow.dev/examples/layout/dagre

How do I refresh the diagram after I have programmatically updated the nodes by calling setNode on the dagre graph.

**Comment by bcakmakoglu** [comment]
What exactly do you mean by refreshing?
Re-layouting?

You could just put all the new nodes and edges through the layouting hook and pass those to setNodes/setEdges, the same way the initial layout is done

**Comment by mjcee** [comment, marked as answer]
Sorry I should have updated the forum entry.

Yes I wanted to update the diagram when I added nodes.  Using setNodes worked for my initial program but didn't seem to work for the Dagre Tree example.  I eventually found the https://reactflow.dev/examples/layout/auto-layout example and that worked with Dagre Tree.

Thanks for your re[ly.
